wrappers/src/notifications_cs.hpp

```diff
@@ -105,9 +106,13 @@ static inline void handle_changes(ManagedNotificationTokenContext* context, Coll
 
         std::vector<int32_t> properties;
 
-        for (auto& pair : changes.columns) {
-            if (!pair.second.empty()) {
-                properties.emplace_back(get_property_index(context->schema, ColKey(pair.first)));
+        //Property names are necessary only for shallow notifications (PropertyChanged/CollectionChanged)
+        if (type == key_path_collection_type::SHALLOW)
+        {
+            for (auto& pair : changes.columns) {
+                if (!pair.second.empty()) {
+                    properties.emplace_back(get_property_index(context->schema, ColKey(pair.first)));
+                }
             }
```

Comment:
Not sure I understand why

Reply:
The comment is not totally correct, it should only be for PropertyChanged. At the moment, we need the property names only to raise `PropertyChanged`, while it's not used for `CollectionChanged` or `SubscribeForNotifcations`

Reply:
Gotcha. Still, I think it's a bit awkward that we're using a seemingly unrelated piece of information to decide to extract the property change info. Yes, property change notifications use shallow, but that's incidental and prone to change.

Reply:
I agree with this, I'll remove the if. When we get to object notifications, we can think about how to make the properties calculations only for objects and not for collections.